kommunicate/src/main/java/io/kommunicate/Kommunicate.java

```diff
@@ -802,8 +801,25 @@ private static String getClientGroupId(String userId, List<String> agentIds, Lis
     }
 
     public static KMUser getVisitor() {
-        KMUser user = new KMUser();
+        final KMUser user = new KMUser();
         user.setUserId(generateUserId());
+        new KmAppSettingTask(ApplozicService.getAppContext(),
+                MobiComKitClientService.getApplicationKey(ApplozicService.getAppContext()),
+                new KmCallback() {
+                    @Override
+                    public void onSuccess(Object message) {
+                        final KmAppSettingModel appSettingModel = (KmAppSettingModel) message;
+                        if (appSettingModel != null && appSettingModel.getResponse() != null && appSettingModel.getChatWidget() != null) {
+                            if (appSettingModel.getChatWidget().isPseudonymsEnabled()) {
+                                user.setDisplayName(appSettingModel.getResponse().getUserName());
```

Comment:
what if `appSettingModel.getResponse().getUserName()` is nil?

Reply:
It should not be null if getChatWidget().isPseudonymsEnabled() is true, but I have added a TextUtils.isEmpty check for safety.